pkg/registry/apis/playlist/storage.go

```diff
@@ -4,52 +4,28 @@ import (
 	"strings"
 
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apiserver/pkg/registry/generic"
 	genericregistry "k8s.io/apiserver/pkg/registry/generic/registry"
 
 	playlist "github.com/grafana/grafana/apps/playlist/apis/playlist/v0alpha1"
+	"github.com/grafana/grafana/pkg/apimachinery/utils"
 	grafanaregistry "github.com/grafana/grafana/pkg/apiserver/registry/generic"
-	grafanarest "github.com/grafana/grafana/pkg/apiserver/rest"
 )
 
-var _ grafanarest.Storage = (*storage)(nil)
-
-type storage struct {
-	*genericregistry.Store
-}
-
-func newStorage(scheme *runtime.Scheme, optsGetter generic.RESTOptionsGetter, legacy *legacyStorage) (*storage, error) {
-	gr := schema.GroupResource{
-		Group:    playlist.PlaylistKind().Group(),
-		Resource: playlist.PlaylistKind().Plural(),
-	}
-	singularGR := schema.GroupResource{
-		Group:    playlist.PlaylistKind().Group(),
-		Resource: strings.ToLower(playlist.PlaylistKind().Kind()),
-	}
-	strategy := grafanaregistry.NewStrategy(scheme, gr.WithVersion(playlist.PlaylistKind().Version()).GroupVersion())
-	store := &genericregistry.Store{
-		NewFunc: func() runtime.Object {
-			return playlist.PlaylistKind().ZeroValue()
+func newStorage(scheme *runtime.Scheme, optsGetter generic.RESTOptionsGetter, legacy *legacyStorage) (*genericregistry.Store, error) {
+	kind := playlist.PlaylistKind()
+	singular := strings.ToLower(kind.Kind()) // ???
+	resourceInfo := utils.NewResourceInfo(
```

Comment:
This one is the most interesting -- seems we should try to align "resourceInfo" and kind.  It could be the ResourceInfo would implement Kind 🤔  and we may want to add some options to kind

Reply:
yeah, agreed